components/device-mgt/org.wso2.carbon.device.mgt.core/src/main/java/org/wso2/carbon/device/mgt/core/DeviceManagementServiceProviderImpl.java

```diff
@@ -571,4 +571,14 @@ public Operation getOperation(int operationId) throws OperationManagementExcepti
         }
         return devicesOfRole;
     }
+
+    @Override
+    public int getDeviceCount() throws DeviceManagementException {
+        try {
+            int deviceCount = this.deviceDAO.getDeviceCount();
```

Comment:
Redundant variable assignment. You can perhaps return the output of "this.deviceDAO.getDeviceCount();" directly.